lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -437,6 +452,62 @@ def get_course_topics(request: Request, course_key: CourseKey, topic_ids: Option
     }
 
 
+def get_course_topics_v2(
+    course_key: CourseKey,
+    user: User,
+    topic_ids: Optional[Iterable[str]] = None,
+    order_by: TopicOrdering = TopicOrdering.COURSE_STRUCTURE,
+) -> List[Dict]:
+    """
+    Returns the course topic listing for the given course and user; filtered
+    by 'topic_ids' list if given.
+
+    Parameters:
+
+        course_key: The key of the course to get topics for
+        user: The requesting user, for access control
+        topic_ids: A list of topic IDs for which topic details are requested
+        order_by: The sort ordering for the returned list of topics
+
+    Returns:
+
+        A list of discussion topics for the course.
+
+    Raises:
+        ValidationError: If unsupported ordering is used.
+    """
+    # Check access to the course
+    _get_course(course_key, user=user, check_tab=False)
+    provider_type = DiscussionsConfiguration.get(context_key=course_key).provider_type
+    topics_query = DiscussionTopicLink.objects.filter(
+        context_key=course_key,
+        provider_id=provider_type,
+    )
+    if topic_ids:
+        topics_query = topics_query.filter(external_id__in=topic_ids)
+
+    if provider_type in [Provider.OPEN_EDX, Provider.LEGACY]:
```

Comment:
How you are specifying these providers? I mean what makes open_edx provider compatible here with a legacy for getting thread_counts? What we can call this set of providers?

Reply:
These are both providers that use the cs_comments_service, so we can call that to get the thread counts for the other providers we don't have any such common implementation.